yescrypt-sha256.h

```diff
@@ -0,0 +1,121 @@
+/*-
```

Comment:
This is duplicate code. The SHA256 hash algorythm is already implemented in `alg-sha256.h`. Implementing HMAC-SHA256 is pretty simple as well. See `alg-hmac-sha1.h` for a reference.

Reply:
It's slightly different in API (like have additional `tmp32` parameter), and implements HMAC_SHA256 on top of this, and PBKDF2_SHA256 on top of that (having calls to things like SHA256_Pad_Almost(), which `alg-sha256.c` does not provide).

Reply:
`SHA256_Pad_Almost()` isn't used in yescrypt code-base anywhere; it's a simple helper function, that is called by this specific implementation of `PBKDF2_SHA256()`.

Implementing this functionality in several seperate reusable files (and adding the official RFC and FIPS-180 test-vectors) shouldn't be to much of a problem.

Reply:
`SHA256_Pad_Almost()` _is_ indirectly used in yescrypt, and I added it specifically for its use in yescrypt. This happens via `PBKDF2_SHA256()` when it's called by yescrypt with c=1, as yescrypt always does (and as scrypt always did). This is optimization for this special case of c=1. While we could simplify things for libxcrypt in particular and lose maybe just ~1% of performance on this (since yescrypt spends most time in its own heavy processing and not in its initial/final PBKDF2 steps), I am uncomfortable maintaining two versions of the code and for some larger deployments a combination of micro-optimizations like this was actually needed to squeeze out the last few missing percent of performance to meet server request rate capacity requirements. An option for libxcrypt is to revert to using Colin's sha256.c from the scrypt tree as-is, though. I maintained compatibility with it.